pkg/autodiscovery/providers/container.go

```diff
@@ -73,7 +75,7 @@ func (d *ContainerConfigProvider) Collect(ctx context.Context) ([]integration.Co
 	return d.generateConfigs()
 }
 
-func (d *ContainerConfigProvider) listen() {
+func (d *ContainerConfigProvider) listen(ch chan struct{}) {
```

Comment:
```suggestion
// listen, closing the given channel after the initial set of events are received
func (d *ContainerConfigProvider) listen(ranOnceCh chan struct{}) {
```
(just for clarity of intention)